cumulusci/oauth/salesforce.py

```diff
@@ -74,6 +78,9 @@ def jwt_session(client_id, private_key, username, url=None, auth_url=None):
     headers = {"Content-Type": "application/x-www-form-urlencoded"}
     token_url = urljoin(url, "services/oauth2/token")
     response = requests.post(url=token_url, data=data, headers=headers)
+    if response.status_code == 400:
+        raise CumulusCIException(ERROR_MESSAGE_400.format(response.text))
```

Comment:
I think it would make sense to use the same logic that we do for checking the response when we do the refresh token flow, and raise the same exception class: https://github.com/SFDO-Tooling/CumulusCI/blob/main/cumulusci/core/config/OrgConfig.py#L106-L109